src/os/bluestore/BlueFS.cc

```diff
@@ -144,7 +144,7 @@ void BlueFS::add_block_extent(unsigned id, uint64_t offset, uint64_t length)
   block_all[id].insert(offset, length);
   block_total[id] += length;
 
-  if (alloc.size()) {
+  if (id < alloc.size() && alloc[id]) {
```

Comment:
Doesn't that mean there's a problem with the UT though?  The test shouldn't be asking for an ID that is outside the bounds of the alloc vector.  Seems like Igor's approach is correct and we must fix the UT if that's the case?